src/transformers/models/whisper/modeling_whisper.py

```diff
@@ -1566,8 +1577,19 @@ def generate(
         if task is not None:
             generation_config.task = task
 
-        forced_decoder_ids = []
-        if task is not None or language is not None:
+        forced_decoder_ids = None
+
+        # Legacy code for backward compatibility
+        if hasattr(self.config, "forced_decoder_ids") and self.config.forced_decoder_ids is not None:
+            forced_decoder_ids = self.config.forced_decoder_ids
+        elif (
+            hasattr(self.generation_config, "forced_decoder_ids")
+            and self.generation_config.forced_decoder_ids is not None
+        ):
+            forced_decoder_ids = self.generation_config.forced_decoder_ids
+
+        if task is not None or language is not None or (forced_decoder_ids is None and prompt_ids is not None):
```

Comment:
This is done solely for handling the case where `prompt_ids` are passed in but the generation config and model config's forced decoder ids are both `None`. Its essentially just changing the order of operations so that we can cleanly check `forced_decoder_ids is None and prompt_ids is not None` to then add non-prompt forced decoder ids, none of the other functionality should change